packages/dd-trace/src/debugger/devtools_client/breakpoints.js

```diff
@@ -43,54 +44,124 @@ async function addBreakpoint (probe) {
     ({ line: lineNumber, column: columnNumber } = await getGeneratedPosition(url, source, lineNumber, sourceMapURL))
   }
 
+  try {
+    probe.condition = probe.when?.json && compileCondition(probe.when.json)
+  } catch (err) {
+    throw new Error(`Cannot compile expression: ${probe.when.dsl}`, { cause: err })
+  }
+
+  const release = await lock()
```

Comment:
Should `try` be moved to L54 to ensure `release()` is always called in the `finally` block? Everything here looks safe, but it could protect against future mistakes.

Reply:
Nice catch (no pun intended)... it is fixed now 👍 It was just before the `try` block in an older version of this PR, but after some refactoring it got moved 😅